packages/react-native/src/modules/push/KnockPushNotificationProvider.tsx

```diff
@@ -0,0 +1,95 @@
+import { ChannelData } from "@knocklabs/client";
+import { useKnockClient } from "@knocklabs/react-core";
+import React, { createContext, useCallback, useContext } from "react";
+
+export interface KnockPushNotificationContextType {
+  registerPushTokenToChannel(token: string, channelId: string): Promise<void>;
+  unregisterPushTokenFromChannel(
+    token: string,
+    channelId: string,
+  ): Promise<void>;
+}
+
+const KnockPushNotificationContext = createContext<
+  KnockPushNotificationContextType | undefined
+>(undefined);
+
+export interface KnockPushNotificationProviderProps {
+  children?: React.ReactElement;
+}
+
+export const KnockPushNotificationProvider: React.FC<
```

Comment:
For what it’s worth, this `KnockPushNotificationProvider` does not use any APIs specific to React Native. Alternatively, we could add this to `@knocklabs/react-core`.

`@knocklabs/react-native` already re-exports everything from `@knocklabs/react-core`, so it would still be available for use via our React Native SDK:

https://github.com/knocklabs/javascript/blob/5d60ed0815a7697bc9a2e59c0ba5b71914e04084/packages/react-native/src/index.ts#L2